contracts/modules/TransferManager/ManualApprovalTransferManager.sol

```diff
@@ -102,9 +148,44 @@ contract ManualApprovalTransferManager is ITransferManager {
     function revokeManualApproval(address _from, address _to) public withPerm(TRANSFER_APPROVAL) {
         require(_to != address(0), "Invalid to address");
         delete manualApprovals[_from][_to];
+
+        //find the record in active approvals array & delete it
+        uint256 index;
+        for(uint256 i = 0; i < activeManualApprovals.length; i++){
```

Comment:
This will go out of gas when a lot of manual approvals are added. Consider saving the index in the struct or somewhere else to directly delete elements. See other places we are deleting restrictions for the pattern of moving last element to current position and deleting last element to preserve indexes.